crates/bitwarden-ipc/src/ipc_client.rs

```diff
@@ -42,170 +47,270 @@ pub struct IpcClientTypedSubscription<Crypto, Com, Ses, Payload>
             sessions,
+
+            incoming: RwLock::new(None),
+            cancellation_handle: RwLock::new(None),
         })
     }
 
+    pub async fn start(self: &Arc<Self>) -> Result<(), StartError> {
+        let client = self.clone();
+        let (cancellation_handle_tx, mut cancellation_handle_rx) =
+            tokio::sync::watch::channel(false);
+        let (await_init_tx, await_init_rx) = tokio::sync::oneshot::channel();
+        let mut cancellation_handle = self.cancellation_handle.write().await;
+
+        *cancellation_handle = Some(cancellation_handle_tx);
+
+        let future = async move {
+            let com_receiver = client.communication.subscribe().await;
+            let (client_tx, client_rx) = tokio::sync::broadcast::channel(CHANNEL_BUFFER_CAPACITY);
+
+            let mut client_incoming = client.incoming.write().await;
+            *client_incoming = Some(client_rx);
+            drop(client_incoming);
+
+            await_init_tx
+                .send(())
+                .expect("Sending init signal should not fail");
+
+            loop {
+                select! {
+                    _ = cancellation_handle_rx.changed() => {
+                        if *cancellation_handle_rx.borrow() {
+                            log::debug!("Cancellation signal received, stopping IPC client");
+                            break;
+                        }
+                    }
+                    received = client.crypto.receive(&com_receiver, &client.communication, &client.sessions) => {
+                        match received {
+                            Ok(message) => {
+                                if client_tx.send(message).is_err() {
+                                    log::error!("Failed to save incoming message");
+                                    break;
+                                };
+                            }
+                            Err(e) => {
+                                log::error!("Error receiving message: {:?}", e);
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            log::debug!("IPC client shutting down");
+            client.stop().await;
+        };
+
+        #[cfg(not(target_arch = "wasm32"))]
+        tokio::spawn(future);
+
+        #[cfg(target_arch = "wasm32")]
+        wasm_bindgen_futures::spawn_local(future);
+
+        await_init_rx.await.map_err(|e| StartError(e.to_string()))?;
+        Ok(())
+    }
+
+    pub async fn is_running(self: &Arc<Self>) -> bool {
+        let incoming = self.incoming.read().await;
+        let cancellation_handle = self.cancellation_handle.read().await;
+        incoming.is_some() && cancellation_handle.is_some()
+    }
+
+    pub async fn stop(self: &Arc<Self>) {
+        let mut incoming = self.incoming.write().await;
+        if let Some(receiver) = incoming.take() {
+            drop(receiver);
+        }
```

Comment:
You can also take the value and assign it to _ to immediately drop it.
```suggestion
        let _ = incoming.take();
```